src/Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions/IConnectionInformation.cs

```diff
@@ -12,9 +12,10 @@ public interface IConnectionInformation
         IPEndPoint LocalEndPoint { get; }
 
         PipeFactory PipeFactory { get; }
+        IScheduler InputReaderScheduler { get; }
```

Comment:
I just discussed this with @geoffkizer, I think we'll end up with a `RequiresDispatch` bool on `IConnectionContext` that will determine the default for a transport's default (stay on the same thread or dispatch to the thread pool). The option on KestrelOptions will override this of course.